src/ty.rs

```diff
@@ -44,6 +46,9 @@ pub enum BuiltinType {
     // some trait
     Trait(TraitId),
 
+    // some module
+    Module(ModuleId),
```

Comment:
Do we really need a type for Module? It's not really clear to me that we really do and what that would mean. I don't think you can assign a module to a variable, so I would assume that we can get away without `BuiltinType::Module`?

Reply:
The general idea is to make a module a value too, and be able to pass them around.
(Not too useful on it's own, but combined with the module extending a parent type, it gets really useful.)

```
module TestConfig : Config;
module ProdConfig : Config;

fun onlyForTesting(config: TestConfig) ->...
fun allowsAllConfigs(config: Config) ->...

onlyForTesting(TestConfig); //ok
onlyForTesting(ProdConfig); //nope
allowsAllConfigs(TestConfig); //ok
allowsAllConfigs(ProdConfig); //nope
```

Reply:
Ok, I see. So it is also possible to use the module-instance as a normal object (e.g. `config.server_name()`, right?

Reply:
But at the same time, I could also use `TestConfig::server_name()`.

Reply:
Yes, exactly!